src/yunohost/settings.py

```diff
@@ -235,3 +243,44 @@ def _save_settings(settings, location=SETTINGS_PATH):
             settings_fd.write(result)
     except Exception as e:
         raise YunohostError('global_settings_cant_write_settings', reason=e)
+
+
+# Meant to be a dict of setting_name -> function to call
+post_change_hooks = {}
+
+
+def post_change_hook(setting_name):
+    def decorator(func):
+        assert setting_name in DEFAULTS.keys(), "The setting %s does not exists" % setting_name
+        assert setting_name not in post_change_hooks, "You can only register one post change hook per setting (in particular for %s)" % setting_name
+        post_change_hooks[setting_name] = func
+        return func
+    return decorator
+
+
+def trigger_post_change_hook(setting_name, old_value, new_value):
+    if setting_name not in post_change_hooks:
+        logger.debug("Nothing to do after changing setting %s" % setting_name)
+        return
+
+    f = post_change_hooks[setting_name]
+    f(old_value, new_value)
+
+
+# ===========================================
+#
+# Actions to trigger when changing a setting
+# You can define such an action with :
+#
+# @post_change_hook("your.setting.name")
+# def some_function_name(old_value, new_value):
```

Comment:
We might need to have the setting name in the parameter of this function.
You could share the same hook between two settings. You could avoid having the setting name hardcoded when you want to log a context for example.

Reply:
Indeed ;) Done !